src/k8s/pkg/k8sd/types/cluster_config_validate.go

```diff
@@ -31,6 +89,14 @@ func (c *ClusterConfig) Validate() error {
 		return fmt.Errorf("invalid service CIDR: %w", err)
 	}
 
+	if err := validateCIDROverlap(c.Network.GetPodCIDR(), c.Network.GetServiceCIDR()); err != nil {
+		return fmt.Errorf("invalid cidr configuration: %w", err)
```

Comment:
nit: maybe we can be more specific with this error (and the one after that)?

Reply:
not really because we don't know the type of error that validatecidroverlap will return... could be a parsing error or a overlap error

Reply:
The actual error is wrapped in the response so we should be fine.